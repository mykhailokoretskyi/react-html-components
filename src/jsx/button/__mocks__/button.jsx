'use strict';

import React from 'react';
import Button from '../button';

export default (<Button >Test button</Button>);
