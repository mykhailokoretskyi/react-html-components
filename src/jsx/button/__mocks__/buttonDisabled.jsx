'use strict';

import React from 'react';
import Button from '../button';

export default (<Button disabled={true}>Test button</Button>);
