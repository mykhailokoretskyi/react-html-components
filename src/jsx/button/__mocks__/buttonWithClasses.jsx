'use strict';

import React from 'react';
import Button from '../button';

export default (<Button classes={['some', 'classes']}>Test button</Button>);
