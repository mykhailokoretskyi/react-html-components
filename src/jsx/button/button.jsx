'use strict';

import React, { Component, PropTypes } from 'react';

export default class Button extends Component {
    constructor(props){
        super(props);

        this._onClick = this._onClick.bind(this);

        this.state = {
            disabled: this.props.disabled
        };
    }

    componentDidUpdate(prevProps, prevState){
        if(this.state.disabled !== prevState.disabled){
            this.props.changeCallback();
        }
    }

    _onClick(e){
        if (!this.state.disabled) {
            this.props.clickCallback();
        }
    }

    getClassName(classes){
        return (classes && classes.length) ? classes.join(' ') : '';
    }

    disabled(v){
        if (typeof v === 'undefined'){
            return this.state.disabled;
        } else {
            this.setState({
                disabled: v
            });
        }
    }

    render(){
        const type = this.props.type ? `btn-${this.props.type}` : 'btn';
        const classes = this.props.classes;
        classes.push(type);
        if (this.state.disabled) {
            classes.push('disabled');
        }
        const className = this.getClassName(classes);
        return (
            <a className={className} onClick={this._onClick} >
                {this.props.children}
            </a>
        );
    }
}

Button.types = [
    '',
    'flat',
    'floating',
    'large'
];

Button.propTypes = {
    type:           PropTypes.oneOf(Button.types),
    disabled:       PropTypes.bool,
    classes:        PropTypes.array,
    changeCallback: PropTypes.func,
    clickCallback:  PropTypes.func
};

Button.defaultProps = {
    type:           '',
    disabled:       false,
    classes:        [],
    changeCallback: function(){},
    clickCallback:  function(){}
};
