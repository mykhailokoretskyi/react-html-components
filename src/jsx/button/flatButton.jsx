'use strict';

import React, { Component } from 'react';
import Button from './button';

export default class FlatButton extends Component {
    constructor(props){
        super(props);
    }

    render(){
        return (
            <Button {...this.props} type="flat">
                { this.props.children }
            </Button>
        );
    }
}
