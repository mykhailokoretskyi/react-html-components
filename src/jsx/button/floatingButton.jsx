'use strict';

import React, { Component } from 'react';
import Button from './button';

export default class FloatingButton extends Component {
    constructor(props){
        super(props);
    }

    render(){
        return (
            <Button {...this.props} type="floating">
                { this.props.children }
            </Button>
        );
    }
}
