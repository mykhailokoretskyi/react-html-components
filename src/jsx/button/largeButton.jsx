'use strict';

import React, { Component } from 'react';
import Button from './button';

export default class LargeButton extends Component {
    constructor(props){
        super(props);
    }

    render(){
        return (
            <Button {...this.props} type="large">
                { this.props.children }
            </Button>
        );
    }
}
