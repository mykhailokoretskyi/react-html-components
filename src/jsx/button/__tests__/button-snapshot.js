jest.mock('react/lib/ReactDefaultInjection');
import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';
import FloatingButton from '../floatingButton';
import FlatButton from '../flatButton';
import LargeButton from '../largeButton';

import button from '../__mocks__/button';
import buttonWithClasses from '../__mocks__/buttonWithClasses';
import buttonDisabled from '../__mocks__/buttonDisabled';

describe('Button', () => {
    it('regular button snapshot', () => {
        const component = renderer.create(button);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('regular with classes', () => {
        const component = renderer.create(buttonWithClasses);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('regular disabled', () => {
        const component = renderer.create(buttonDisabled);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('large matches snapshot', () => {
        const component = renderer.create(<LargeButton classes={["some", "classes"]}>Large</LargeButton>);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('flat matches snapshot', () => {
        const component = renderer.create(<FlatButton classes={["some", "classes"]}>Flat</FlatButton>);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('floating matches snapshot', () => {
        const component = renderer.create(<FloatingButton classes={["some", "classes"]}>Floating</FloatingButton>);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
