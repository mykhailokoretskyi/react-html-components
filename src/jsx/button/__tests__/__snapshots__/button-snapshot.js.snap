exports[`Button flat matches snapshot 1`] = `
<a
  className="some classes btn-flat"
  onClick={[Function bound _onClick]}>
  Flat
</a>
`;

exports[`Button floating matches snapshot 1`] = `
<a
  className="some classes btn-floating"
  onClick={[Function bound _onClick]}>
  Floating
</a>
`;

exports[`Button large matches snapshot 1`] = `
<a
  className="some classes btn-large"
  onClick={[Function bound _onClick]}>
  Large
</a>
`;

exports[`Button regular button snapshot 1`] = `
<a
  className="btn"
  onClick={[Function bound _onClick]}>
  Test button
</a>
`;

exports[`Button regular disabled 1`] = `
<a
  className="btn btn disabled"
  onClick={[Function bound _onClick]}>
  Test button
</a>
`;

exports[`Button regular with classes 1`] = `
<a
  className="some classes btn"
  onClick={[Function bound _onClick]}>
  Test button
</a>
`;
